### PR TITLE
Add tests of "withSequence" and clean up imports

### PR DESCRIPTION
**Pull request by Latropos**
## Summary
Add tests of withSequence and clean up the import structure of tests

### CODE REVIEW

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/arrays.test.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        <Animated.View ref={ref1} style={[styles.animatedBox, style1]} />
+        <Animated.View ref={ref2} style={[styles.animatedBox, style2]} />
+      </View>
+    );
+  };
+
+  test.each([
+    { startValues: [0, 10, 20], middleValues: [0, 10, 20], finalValues: [20, 10, 30], animationNumber: 0 },
+    { startValues: [0, 10, 20], middleValues: [0, 50, 60], finalValues: [40, 10, 30], animationNumber: 1 },
+    { startValues: [30, 10, 55], middleValues: [0, -10, 60], finalValues: [40, 10, 30], animationNumber: 1 },
```
Comment (latekvo) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/arrays.test.tsx:98]: Shouldn't this be `animationNumber: 2`?
Width component has cases for animation numbers: `0`, `1`, and `2`, but only `0` and `1` is used.
If it's intentional, is it necessary to have a `case` for `animationNumber = 2` in the `WidthComponent`?

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    ACTIVE = 'ONE',
+    PASSIVE = 'TWO',
+  }
+  const DELAY = 75;
+  const WidthComponent = ({ startColor, middleColor, finalColor }: TestCase) => {
+    const colorActiveSV = useSharedValue(startColor);
+    const colorPassiveSV = useSharedValue(startColor);
+
+    const refOne = useTestRef(Component.ACTIVE);
+    const refTwo = useTestRef(Component.PASSIVE);
```
Comment (latekvo) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx:38]: Wouldn't it be better to name those refs according to what they're holding?
e.g `refActive`, `refPassive`

**Review by latekvo — COMMENTED** [review]
Looks great, just those 2 minor questions.

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/arrays.test.tsx (hunk 1 above)
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/arrays.test.tsx:98]:
```suggestion
    { startValues: [30, 10, 55], middleValues: [0, -10, 60], finalValues: [40, 10, 30], animationNumber: 2 },
```

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      finalColor: 'hsl(120,100%,50%)',
+      finalColorHex: '#00ff00ff',
+    },
+    {
+      startColor: 'hwb(70, 50%, 0%)',
+      startColorHex: '#eaff80ff',
+      middleColor: 'hsl(180, 50%, 50%)',
+      middleColorHex: '#40bfbfff',
+      finalColor: 'hsl(120,100%,50%)',
+      finalColorHex: '#00ff00ff',
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx:106]: could you also add some tests for `rgb`/`rgba` colors?

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    },
+  ])(
+    'Animate ${startColor} → ${finalColor} → ${middleColor} → ${finalColor}',
+    async ({ startColor, startColorHex, middleColor, middleColorHex, finalColor, finalColorHex }) => {
+      await render(<WidthComponent startColor={startColor} middleColor={middleColor} finalColor={finalColor} />);
+      const activeComponent = getTestComponent(Component.ACTIVE);
+      const passiveComponent = getTestComponent(Component.PASSIVE);
+
+      await wait(DELAY / 2);
+      // TODO Decide what should be the starting value of activeComponent
```
Comment (piaskowyk) [inline apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx:116]: Do you have some a list of issues that you found?

**Review by piaskowyk — COMMENTED** [review]
Very solid 👍

File: apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx (hunk 4 above)
Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx:116]: As you can see there are two components that should behave the same way, but they don't:
```tsx
expect(await activeComponent.getAnimatedStyle('backgroundColor'))
    .not.toBe(startColorHex, ComparisonMode.COLOR);
expect(await passiveComponent.getAnimatedStyle('backgroundColor'))
    .toBe(startColorHex, ComparisonMode.COLOR);
```

The problem with the activeComponent that the following code:
```diff
!   const colorActiveSV = useSharedValue(startColor);

    const styleActive = useAnimatedStyle(() => {
      return {
        backgroundColor: withSequence(
!         withDelay(DELAY, withTiming(colorActiveSV.value, { duration: 200 })),
          withDelay(DELAY, withTiming(middleColor, { duration: 300 })),
          withDelay(DELAY, withTiming(colorActiveSV.value, { duration: 200 })),
        ),
      };
    });

    useEffect(() => {
!     colorActiveSV.value = finalColor;
    }, [colorActiveSV, finalColor]);
 ```
 Should produce the animation: startColor → finalColor → middleColor → finalColor,
 but we get finalColor → middleColor → finalColor

Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx:116]: Thats the way it looks:

https://github.com/software-mansion/react-native-reanimated/assets/56199675/fe692b5e-d9f9-4c24-8639-580d7c0f541d

Comment (Latropos) [reply apps/common-app/src/examples/RuntimeTests/tests/animations/withSequence/colors.test.tsx:116]: I was checking several times if this should be considered a bug and I think it should - when we have similar ways to animate with other animations, we get the bottom result,

**Review by latekvo — APPROVED** [review]

**Pull request merged**